PresentScreenings/Menu Delegates/FilmsMenuDelegate.cs

```diff
@@ -105,5 +122,104 @@ public override void NeedsUpdate(NSMenu menu)
+            {
+                return;
+            }
+
+            // Remove current extra menu items.
+            foreach (nint tag in _enabledByTag.Keys)
+            {
+                menu.RemoveItem(menu.ItemWithTag(tag));
+            }
+            _enabledByTag = new Dictionary<nint, bool> { };
+
+            // Get the controller.
+            FilmRatingDialogController controller = _app.FilmsDialogController;
+
+            // Create extra menu items if applicable.
+            if (controller != null)
+            {
+                Film film = controller.CurrentFilm;
+                if (film != null)
+                {
+                    FilmInfo filmInfo = ViewController.GetFilmInfo(film.FilmId);
+                    if (filmInfo != null)
+                    {
+                        CreateAllExtraMenuItems(menu, filmInfo);
+                    }
+                }
+            }
+        }
+
+        private void CreateAllExtraMenuItems(NSMenu menu, FilmInfo filmInfo)
+        {
+            _currentTag = _firstExtraMenuItemTag;
+
+            // Create the screened films items.
+            if (filmInfo.ScreenedFilms.Count > 0)
+            {
+                var screenedFilms = filmInfo.ScreenedFilms
+                    .Select(sf => ViewController.GetFilmById(sf.ScreenedFilmId))
+                    .ToList();
+                CreateExtraMenuItems(menu, screenedFilms, "Screened Films");
+            }
+
+            // Create the combination program items.
+            if (filmInfo.CombinationProgramIds.Count > 0)
+            {
+                var combinationFilms = filmInfo.CombinationProgramIds
+                    .Select(id => ViewController.GetFilmById(id))
+                    .ToList();
+                CreateExtraMenuItems(menu, combinationFilms, "Combination Program(s)");
+            }
+        }
+
+        private void CreateExtraMenuItems(NSMenu menu, List<Film> extraFilms, string headerTitle)
+        {
+            // Create the separator item.
+            _currentTag += 1;
+            NSMenuItem seperatorItem = NSMenuItem.SeparatorItem;
+            seperatorItem.Tag = _currentTag;
+            _enabledByTag.Add(_currentTag, false);
+            menu.AddItem(seperatorItem);
+
+            // Create the header item.
+            _currentTag += 1;
+            NSMenuItem headerItem = new NSMenuItem
+            {
+                Title = headerTitle,
+                Tag = _currentTag,
+                IndentationLevel = 1,
+            };
+            _enabledByTag.Add(_currentTag, false);
+            menu.AddItem(headerItem);
+
+            // Create an item for each extra film.
+            NSEventModifierMask mask = NSEventModifierMask.ControlKeyMask | NSEventModifierMask.CommandKeyMask;
+            int extraFilmNumber = 0;
+            foreach (var extraFilm in extraFilms)
+            {
+                _currentTag += 1;
+                extraFilmNumber += 1;
```

Comment:
Please consider to guarantee that different menu items cannot receive the same key equivalent.

Reply:
Done.